src/main/java/org/grouplens/grapht/reflect/internal/ClassInstantiator.java

```diff
@@ -74,20 +78,24 @@ public Class getType() {
     @Override
     public Object instantiate() throws ConstructionException {
         // find constructor and build up necessary constructor arguments
+
+        LogContext mdcContext = LogContext.create();
         Constructor<?> ctor = getConstructor();
-        Object[] ctorArgs = new Object[ctor.getParameterTypes().length];
-        for (Desire d: desires) {
-            if (d.getInjectionPoint() instanceof ConstructorParameterInjectionPoint) {
-                // this desire is a constructor argument so create it now
-                Instantiator provider = providers.get(d);
-                ConstructorParameterInjectionPoint cd = (ConstructorParameterInjectionPoint) d.getInjectionPoint();
-                ctorArgs[cd.getParameterIndex()] = checkNull(cd, provider.instantiate());
-            }
-        }
-        
         // create the instance that we are injecting
         Object instance;
         try {
+            mdcContext.put("org.grouplens.grapht.class", ctor.getClass().toString());
+            Object[] ctorArgs = new Object[ctor.getParameterTypes().length];
+            for (Desire d : desires) {
+                if (d.getInjectionPoint() instanceof ConstructorParameterInjectionPoint) {
+                    // this desire is a constructor argument so create it now
+                    Instantiator provider = providers.get(d);
+                    ConstructorParameterInjectionPoint cd = (ConstructorParameterInjectionPoint) d.getInjectionPoint();
+                    logger.trace("Injection point satisfactions in progress {}",cd);
+                    mdcContext.put("org.grouplens.grapht.class.injectionpoint", cd.toString());
+                    ctorArgs[cd.getParameterIndex()] = checkNull(cd, provider.instantiate());
+                }
+            }
```

Comment:
I think this is OK. I'm pretty sure none of this code can thrown one of the exceptions we're catching.